pdns/dnsdistdist/dnsdist-backend.cc

```diff
@@ -905,10 +905,10 @@ void DownstreamState::registerXsk(std::vector<std::shared_ptr<XskSocket>>& xsks)
   d_config.sourceMACAddr = d_xskSockets.at(0)->getSourceMACAddress();
 
   for (auto& xsk : d_xskSockets) {
-    auto xskInfo = XskWorker::create();
+    auto xskInfo = XskWorker::create(XskWorker::Type::Bidirectional);
```

Comment:
In the cases I have seen, all `XskWorker::create()` calls are immediately followed by a call to `xskInfo->setSharedFrames()`. Here there's some work going on in between, is this safe? Would it be possible to change the constructor so it takes both a `Type` and a `std::shared_ptr<LockGuarded<vector<uint64_t>>>&)`, and do away with the separate call to `setSharedFrames()`?

Reply:
Done, thanks!